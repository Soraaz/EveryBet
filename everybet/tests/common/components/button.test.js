import React from "react";
import {TouchableOpacity} from "react-native";
import renderer from "react-test-renderer";
import {shallow} from "enzyme";

import Button from "../../../src/common/components/button";

describe("Button", () => {
	describe("rendering", () => {
		it("Should render the normal button", () => {
			const tree = renderer
				.create(
					<Button
						onPress={() => {}}
						type={"normal"}
						text={"text"}
					/>
				)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render the small normal button", () => {
			const tree = renderer
				.create(
					<Button
						onPress={() => {}}
						type={"normalSmall"}
						text={"text"}
					/>
				)				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render the success button", () => {
			const tree = renderer
				.create(
					<Button
						onPress={() => {}}
						type={"success"}
						text={"text"}
					/>
				)				.toJSON();
			expect(tree).to.matchSnapshot();
		});
		it("Should render the warning button", () => {
			const tree = renderer
				.create(
					<Button
						onPress={() => {}}
						type={"warning"}
						text={"text"}
					/>
				)				.toJSON();
			expect(tree).to.matchSnapshot();
		});
		it("Should render the send button", () => {
			const tree = renderer
				.create(
					<Button
						onPress={() => {}}
						type={"send"}
						text={"text"}
					/>
				)				.toJSON();
			expect(tree).to.matchSnapshot();
		});
	});

	describe("functionalities", () =>{
		let onPress;
		let wrapper;

		beforeEach(() => {
			onPress = sinon.spy();
			wrapper = shallow(
				<Button
					onPress={onPress}
					type={"normal"}
					text={"text"}
				/>
			);
		});

		it("Should call the callback function onPress", () => {
			expect(onPress).to.be.not.called;
			wrapper.find(TouchableOpacity).first().props().onPress();
			expect(onPress).to.be.calledOnce;
		});
	});
});
