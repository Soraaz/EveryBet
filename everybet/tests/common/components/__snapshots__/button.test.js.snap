// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button rendering Should render the normal button 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "height": 45,
      "justifyContent": "center",
      "marginTop": 20,
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 7.5,
      "width": 215,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    text
  </Text>
</View>
`;

exports[`Button rendering Should render the send button 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#C9ECE9",
      "borderRadius": 10,
      "height": 45,
      "justifyContent": "center",
      "marginTop": 20,
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 7.5,
      "width": 215,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    text
  </Text>
</View>
`;

exports[`Button rendering Should render the small normal button 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "height": 30,
      "justifyContent": "center",
      "marginTop": 20,
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 7.5,
      "width": 171,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    text
  </Text>
</View>
`;

exports[`Button rendering Should render the success button 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#A6E48D",
      "borderRadius": 10,
      "height": 45,
      "justifyContent": "center",
      "marginTop": 20,
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 7.5,
      "width": 215,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    text
  </Text>
</View>
`;

exports[`Button rendering Should render the warning button 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#FC8D8D",
      "borderRadius": 10,
      "height": 45,
      "justifyContent": "center",
      "marginTop": 20,
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 7.5,
      "width": 215,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    text
  </Text>
</View>
`;
